--- node_modules/sodium-browserify/index.js~	1985-10-26 09:15:00.000000000 +0100
+++ node_modules/sodium-browserify/index.js	2019-07-31 12:00:37.728121669 +0200
@@ -2,7 +2,27 @@
 //to get automatic fallbacks!
 
 //load tweetnacl first, so that it works sync, and everything is there.
-var exports = require('sodium-browserify-tweetnacl')
+var exports = {
+  crypto_auth: function() {},
+  crypto_auth_verify: function() {},
+  crypto_box_easy: function() {},
+  crypto_box_keypair: function() {},
+  crypto_box_open_easy: function() {},
+  crypto_hash: function() {},
+  crypto_hash_sha256: function() {},
+  crypto_scalarmult: function() {},
+  crypto_secretbox_easy: function() {},
+  crypto_secretbox_open_easy: function() {},
+  crypto_sign: function() {},
+  crypto_sign_detached: function() {},
+  crypto_sign_ed25519_pk_to_curve25519: function() {},
+  crypto_sign_ed25519_sk_to_curve25519: function() {},
+  crypto_sign_keypair: function() {},
+  crypto_sign_open: function() {},
+  crypto_sign_seed_keypair: function() {},
+  crypto_sign_verify_detached: function() {},
+  randombytes: function() {}
+}
 
 var EventEmitter = require('events')
 module.exports.events = new EventEmitter()
