--- node_modules/ssb-ebt/index.js~	1985-10-26 09:15:00.000000000 +0100
+++ node_modules/ssb-ebt/index.js	2019-12-20 21:43:16.742065771 +0100
@@ -120,7 +120,7 @@
 
 
   sbot.on('rpc:connect', function (rpc, isClient) {
-    if(isClient) {
+    if(isClient && !SSB.isInitialSync) {
       var opts = {version: 3}
       var a = toPull.duplex(ebt.createStream(rpc.id, opts.version, true))
       var b = rpc.ebt.replicate(opts, function (err) {
