--- node_modules/ssb-friends/index.js~	1985-10-26 09:15:00.000000000 +0100
+++ node_modules/ssb-friends/index.js	2019-09-09 21:41:08.629391436 +0200
@@ -41,30 +41,6 @@
     })
   }
 
-  //opinion: do not authorize peers blocked by this node.
-  sbot.auth.hook(function (fn, args) {
-    var self = this
-    isBlocking({source: sbot.id, dest: args[0]}, function (err, blocked) {
-      if(blocked)
-        args[1](new Error('client is blocked'))
-      else fn.apply(self, args)
-    })
-  })
-
-  if(!sbot.replicate)
-    throw new Error('ssb-friends expects a replicate plugin to be available')
-
-  // opinion: replicate with everyone within max hops (max passed to layered above ^)
-  pull(
-    layered.hopStream({live: true, old: true}),
-    pull.drain(function (data) {
-      if(data.sync) return
-      for(var k in data) {
-        sbot.replicate.request(k, data[k] >= 0)
-      }
-    })
-  )
-
   require('./contacts')(sbot, layered.createLayer, config)
 
   var legacy = require('./legacy')(layered)
