--- node_modules/flumeview-level/index.js~	1985-10-26 09:15:00.000000000 +0100
+++ node_modules/flumeview-level/index.js	2019-07-11 16:49:56.052711025 +0200
@@ -45,7 +45,8 @@
       })
     }
 
-    mkdirp(path.join(dir, name), function () {
+
+    function aftermkdirp() {
       if(closed) return
       db = create()
       db.get(META, {keyEncoding: 'utf8'}, function (err, value) {
@@ -61,7 +62,12 @@
           })
         }
       })
-    })
+    }
+    
+    if (process.title == 'node')
+      mkdirp(path.join(dir, name), aftermkdirp)
+    else
+      aftermkdirp()
 
     return {
       since: since,
