--- ../dominic-sodium-browserify/index.js	2019-07-09 22:37:08.889331405 +0200
+++ node_modules/sodium-browserify/index.js	2019-07-09 22:33:38.521145239 +0200
@@ -1,9 +1,30 @@
-
 //only exports browser api. use chloride module
 //to get automatic fallbacks!
 
 //load tweetnacl first, so that it works sync, and everything is there.
-var exports = require('sodium-browserify-tweetnacl')
+//var exports = require('sodium-browserify-tweetnacl')
+
+var exports = {
+  crypto_auth: function() {},
+  crypto_auth_verify: function() {},
+  crypto_box_easy: function() {},
+  crypto_box_keypair: function() {},
+  crypto_box_open_easy: function() {},
+  crypto_hash: function() {},
+  crypto_hash_sha256: function() {},
+  crypto_scalarmult: function() {},
+  crypto_secretbox_easy: function() {},
+  crypto_secretbox_open_easy: function() {},
+  crypto_sign: function() {},
+  crypto_sign_detached: function() {},
+  crypto_sign_ed25519_pk_to_curve25519: function() {},
+  crypto_sign_ed25519_sk_to_curve25519: function() {},
+  crypto_sign_keypair: function() {},
+  crypto_sign_open: function() {},
+  crypto_sign_seed_keypair: function() {},
+  crypto_sign_verify_detached: function() {},
+  randombytes: function() {}
+}
 
 for(var k in exports) (function (k) {
   if('function' == typeof exports[k])
@@ -15,11 +36,15 @@
     }
 })(k)
 
+var EventEmitter = require('events')
+module.exports.events = new EventEmitter()
+
 //now load wasm which has to be async, ugh.
 var libsodium = require('libsodium-wrappers')
 libsodium.ready.then(function (value, what) {
   require('./browser') (libsodium, exports)
   //set module.exports so that it 
+  module.exports.events.emit("sodium-browserify:wasm loaded")
   module.exports = exports
 }).catch(function (err) {
   //escape from promise land, ugh
