--- node_modules/ssb-validate/index.js~	1985-10-26 09:15:00.000000000 +0100
+++ node_modules/ssb-validate/index.js	2019-07-12 21:12:20.173795898 +0200
@@ -114,6 +114,43 @@
   return err
 }
 
+exports.checkInvalidOOO = function(msg, hmac_key) {
+  if(!ref.isFeedId(msg.author))
+    return new Error('invalid message: must have author')
+  if(!isSigMatchesCurve(msg))
+    return new Error('invalid message: signature type must match author type')
+  if('number' !== typeof msg.timestamp)
+    return fatal(new Error('message must have timestamp, on feed:'+msg.author))
+  if(!isValidOrder(msg, true))
+    return fatal(new Error('message must have keys in allowed order'))
+  if (isInvalidShape(msg))
+    return fatal(new Error('message has invalid shape'))
+  if(!ssbKeys.verifyObj({public: msg.author.substring(1)}, hmac_key, msg))
+    return fatal(new Error('invalid signature'))
+
+  return false // ok
+}
+
+exports.appendOOO = function(state, hmac_key, msg) {
+  if (state.error = exports.checkInvalidOOO(msg, hmac_key))
+    return state
+
+  var kvt = exports.toKeyValueTimestamp(msg)
+
+  if (!state.feeds[msg.author]) {
+    state.feeds[msg.author] = {
+      id: kvt.key,
+      sequence: msg.sequence,
+      timestamp: msg.timestamp,
+      queue: []
+    }
+  }
+
+  state.queue.push(kvt)
+  state.validated += 1
+  return state
+}
+
 exports.checkInvalidCheap = function (state, msg) {
   //the message is just invalid
   if(!ref.isFeedId(msg.author))
